code/game/objects/items/shields.dm

```diff
@@ -330,4 +330,51 @@
 	playsound(src, 'sound/weapons/batonextend.ogg', 50, TRUE)
 	return COMPONENT_NO_DEFAULT_MESSAGE
 
+/obj/item/shield/pillow
+	name = "pillow shield"
+	desc = "A large and tough pillow designed to use in a pillow fight."
+	icon_state = "pillow_shield"
+	inhand_icon_state = "pillow_shield"
+	block_chance = 10
+	max_integrity = 10
+	throw_range = 7
+	throw_speed = 1
+	force = 10
+	damtype = STAMINA
+	w_class = WEIGHT_CLASS_NORMAL
+	var/list/countered_object = list(/obj/item/spear/pillow, /obj/item/shield/pillow, /obj/item/pillow)
+
+/obj/item/shield/pillow/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/boomerang, throw_range, TRUE)
+	AddComponent(/datum/component/two_handed, \
+		force_unwielded = force, \
+		force_wielded = force, \
+	)
+
+/obj/item/shield/pillow/attack(mob/living/target_mob, mob/living/user, params)
+	. = ..()
+	new /obj/effect/temp_visual/pillow_hit(target_mob)
+	playsound(target_mob, 'sound/items/pillow_hit2.ogg', 80)
+
+/obj/item/shield/pillow/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	var/caught = hit_atom.hitby(src, skipcatch = FALSE, hitpush = FALSE, throwingdatum = throwingdatum)
```

Comment:
`.` is `TRUE` if it was caught, fairly certain you should not be calling `hitby` again.